Source/Core/Common/Assembler/GekkoLexer.cpp

```diff
@@ -182,6 +183,10 @@ std::optional<T> EvalIntegral(TokenType tp, std::string_view val)
     return std::accumulate(val.begin() + 2, val.end(), T{0}, bin_step);
   case TokenType::GPR:
   case TokenType::FPR:
+    if (CaseInsensitiveEquals(val, "sp"))
+      val = "r1";
+    if (CaseInsensitiveEquals(val, "rtoc"))
+      val = "r2";
```

Comment:
Instead of re-assigning `val` for `dec_step` to convert the string to a value, why not just return `T{1}` and `T{2}` for "sp" and "rtoc" respectively?

Reply:
It makes the meaning immediately obvious to anyone reading the code, but an early return could work too.

Reply:
Oh, I thought it needed to be returned in a similar fashion as the rest of the returns with std::accumulate and dec_step. Sorry for the misunderstanding